client/src/App/ProtectedRoute.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import {
+  Route,
+  Redirect,
+} from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { useAuth } from './use-auth';
+
+export function ProtectedRoute({ children, ...rest }) {
+  const { user } = useAuth();
```

Comment:
Review for both this line and your `useAuth` in general.

A. It's smart to have a hook/provider from which you fetch the data from.
B. Although, functionality like Login, Signup, and Logout, do not belong in a hook.
C. This way, you have to refetch the same user data from every component that needs it, you already have it once, why fetch it again? This is why Context is a better solution to your issue of providing the user, the rest can simply be functions which are called when needed, no hooks needed.
D. In your components in general, and in your useAuth in specific, always have the bare minimum amount of functions/constants declared, and whenever possible, have functions declared outside of the component. Why? Imagine you're calling this hook/component in 1000 other places (Not an unreasonable number when you have it called inside a loop), then these declarations would take 1000 times more memory than it could've.
E. Don't name components/hooks/hocs in sneak case.

Reply:
Update, I didn't see the context used in the file earlier, it's a good addition, but your use of the hook is still redundent. You could've simply used `ProvideAuth`.